segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/StorageMetadata.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.metadata;
+
+import java.io.Serializable;
+
+/**
+ * Storage Metadata.
+ */
+public interface StorageMetadata extends Serializable {
+
+    /**
+     * Retrieves the key associated with the metadata.
+     * @return key.
+     */
+    String getKey();
+
+    /**
+     * Creates a copy of this instance.
+     * @return
+     */
+    StorageMetadata copy();
```

Comment:
Why do all these classes have a `copy` method?

Reply:
renamed to deepCopy